speech/google/cloud/speech/client.py

```diff
@@ -46,26 +46,17 @@ class Client(client_module.Client):
 
     _connection_class = Connection
 
-    def async_recognize(self, content, source_uri, encoding, sample_rate,
+    def async_recognize(self, encoding, sample_rate, content=None,
```

Comment:
@tseaver yeah I think that would solve that conundrum. Does that depart from trying to keep the API surface simple at all?

Would it be better to just have one `resource` parameter and then check if it `startswith('gs://')`?